HW0/P4/P4.py

```diff
@@ -30,47 +30,97 @@
     # Load true trajectory and plot it
     # Normally, this data wouldn't be available in the real world
     #####################
+    s_true=np.loadtxt('P4_trajectory.txt',delimiter=',')
+    x_coords=s_true[:,0]
+    y_coords=s_true[:,1]
+    z_coords=s_true[:,2]
 
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '--b', label='True trajectory')
+    ax.plot(x_coords, y_coords, z_coords,
+             '--b', label='True trajectory')
 
     #####################
     # Part 2:
     #
     # Read the observation array and plot it (Part 2)
     #####################
-
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '.g', label='Observed trajectory')
+    measurements=np.loadtxt('P4_measurements.txt',delimiter=',')
+    C=np.zeros((3,6))
+    C[0,0]=1/rx
+    C[1,1]=1/ry
+    C[2,2]=1/rz
+    appro=np.dot(measurements,C)
+    x_coords=appro[:,0]
+    y_coords=appro[:,1]
+    z_coords=appro[:,2]
+    ax.plot(x_coords, y_coords, z_coords,
+            '.g', label='Observed trajectory')
 
     #####################
     # Part 3:
     # Use the initial conditions and propagation matrix for prediction
     #####################
 
-    # A = ?
-    # a = ?
-    # s = ?
+    A = np.matrix([[1,0,0,dt,0,0], [0,1,0,0,dt,0], [0,0,1,0,0,dt], 
```

Comment:
Thanks Bryan, I will be sure to do that next time. I actually had trouble typing this up too. That way I am less prone to making mistakes.